java/statestore/src/test/java/sleeper/statestore/dynamodb/DynamoDBStateStoreIT.java

```diff
@@ -958,79 +878,39 @@ public void shouldUpdatePartitions() throws StateStoreException {
+        PartitionTree tree = new PartitionsBuilder(schema)
+                .rootFirst("root")
+                .splitToNewChildren("root", "child1", "child2", 0L)
+                .buildTree();
+        dynamoDBStateStore.initialise(tree.getAllPartitions());
 
         // Then
         assertThat(dynamoDBStateStore.getAllPartitions())
-                .containsExactlyInAnyOrder(parentPartition, childPartition1, childPartition2);
+                .containsExactlyInAnyOrderElementsOf(tree.getAllPartitions());
     }
 
     @Test
     public void shouldNotUpdatePartitionsIfLeafStatusChanges() throws StateStoreException {
         // Given
         Field field = new Field("key", new LongType());
         Schema schema = Schema.builder().rowKeyFields(field).build();
-        RangeFactory rangeFactory = new RangeFactory(schema);
         StateStore dynamoDBStateStore = getStateStore(schema);
-        Partition parentPartition = dynamoDBStateStore.getAllPartitions().get(0);
-        parentPartition = parentPartition.toBuilder()
-                .leafPartition(false)
-                .childPartitionIds(Arrays.asList("child1", "child2"))
-                .build();
-        Region region1 = new Region(rangeFactory.createRange(field, Long.MIN_VALUE, 0L));
-        Partition childPartition1 = Partition.builder()
-                .rowKeyTypes(new LongType())
-                .leafPartition(true)
-                .id("child1")
-                .region(region1)
-                .childPartitionIds(new ArrayList<>())
-                .parentPartitionId(parentPartition.getId())
-                .build();
-        Region region2 = new Region(rangeFactory.createRange(field, 0L, null));
-        Partition childPartition2 = Partition.builder()
-                .rowKeyTypes(new LongType())
-                .leafPartition(true)
-                .id("child2")
-                .region(region2)
-                .childPartitionIds(new ArrayList<>())
-                .parentPartitionId(parentPartition.getId())
-                .build();
-        dynamoDBStateStore.atomicallyUpdatePartitionAndCreateNewOnes(parentPartition, childPartition1, childPartition2);
+
+        PartitionTree tree = new PartitionsBuilder(schema)
+                .rootFirst("root")
+                .splitToNewChildren("root", "child1", "child2", 0L)
+                .buildTree();
+
+        dynamoDBStateStore.initialise(tree.getAllPartitions());
 
         // When / Then
         //  - Attempting to split something that has already been split should fail
-        Partition finalParentPartition = parentPartition;
         assertThatThrownBy(() ->
-                dynamoDBStateStore.atomicallyUpdatePartitionAndCreateNewOnes(finalParentPartition, childPartition1, childPartition2))
+                dynamoDBStateStore.atomicallyUpdatePartitionAndCreateNewOnes(tree.getAllPartitions().get(0), tree.getAllPartitions().get(1), tree.getAllPartitions().get(2)))
```

Comment:
It would be clearer here to use 

`tree.getPartition("root"), tree.getPartition("child1"), tree.getPartition("child2")`

Reply:
I think that's also necessary to make this test valid. The partitions aren't in any particular order from getAllPartitions. The test will still pass incorrectly if the wrong partitions are passed in, because the validation will fail.